test/unit/core/src/ThreadPool.cpp

```diff
@@ -16,6 +22,10 @@ TEST_CASE("threadpool", "[core]")
 
     CHECK_THROWS_AS(f1.get(), std::runtime_error);
 
+#ifdef TSAN
+#    warning "Part of the threadpool test fails the TSAN CI and is therefore disabled."
+    // TODO(bgruber): Revisit this in the future on a new CI image. This problem does not happen locally.
```

Comment:
Interesting. So probably some Ubuntu 20.04 issue then?

Reply:
I have no idea. It does not happen on my Ubuntu 23.04. Also interestingly, the CI reports a higher `BOOST_LIB_STD_GNU` than my local run :D In any case, all I can do is disable the test and add this note. Let's have a look again on newer Ubuntu images.